src/Microsoft.ML.Transforms/SvmLight/SvmLightLoader.cs

```diff
@@ -0,0 +1,851 @@
+                        // Beyond this point _currReader will return null from ReadLine.
+                    }
+                    else
+                        _currReader = _parent._files.OpenTextReader(_fileIdx);
+                    if (_isActive)
+                        _getter = Getter;
+                }
+
+                private void Getter(ref ReadOnlyMemory<char> val)
+                {
+                    Ch.Check(IsGood, "cannot call getter with cursor in its current state");
+                    Ch.Assert(_isActive);
+                    val = _text;
+                }
+
+                public override ValueGetter<TValue> GetGetter<TValue>(DataViewSchema.Column column)
+                {
+                    Ch.CheckParam(column.Index == 0, nameof(column));
+                    Ch.CheckParam(_isActive, nameof(column), "requested column not active");
+                    ValueGetter<TValue> getter = _getter as ValueGetter<TValue>;
+                    if (getter == null)
+                        throw Ch.Except("Invalid TValue: '{0}'", typeof(TValue));
+                    return getter;
+                }
+
+                public override ValueGetter<DataViewRowId> GetIdGetter()
+                {
+                    return
+                        (ref DataViewRowId val) =>
+                        {
+                            Ch.Check(IsGood, "Cannot call ID getter in current state");
+                            val = new DataViewRowId((ulong)Position, 0);
+                        };
+                }
+
+                public override bool IsColumnActive(DataViewSchema.Column column)
+                {
+                    Ch.CheckParam(column.Index == 0, nameof(column));
+                    return _isActive;
+                }
+
+                protected override bool MoveNextCore()
+                {
+                    Ch.AssertValue(_currReader);
+                    Ch.Assert(-1 <= _fileIdx && _fileIdx < (_parent._files == null ? 0 : _parent._files.Count));
+
+                    var count = _parent._files == null ? 0 : _parent._files.Count;
+                    for (; ; )
+                    {
+                        var line = _currReader.ReadLine();
+                        if (line != null)
+                        {
+                            if (_isActive)
+                                _text = line.AsMemory();
+                            return true;
+                        }
+                        if (++_fileIdx == count)
+                            return false;
+                        Ch.AssertValue(_parent._files);
+                        _currReader = _parent._files.OpenTextReader(_fileIdx);
+                    }
+                }
+            }
+        }
+
+        private readonly IHost _host;
+        private readonly ITransformer _keyVectorsToIndexVectors;
+        private readonly FeatureIndices _indicesKind;
+        private readonly ulong _featureCount;
```

Comment:
>_featureCount [](start = 31, length = 13)

Is it useful to make this public?

Reply:
I am not sure there is a need to do that. You can get the length of the feature vector when you call `GetOutputSchema()` on the loader.

---
In reply to: [332212933](https://github.com/dotnet/machinelearning/pull/4190#discussion_r332212933) [](ancestors = 332212933)

Reply:
I'll mark this as resolved, if you think it's important to make public, please re-activate and let me know.

---
In reply to: [333650195](https://github.com/dotnet/machinelearning/pull/4190#discussion_r333650195) [](ancestors = 333650195,332212933)